src/__tests__/data/match-path-data.ts

```diff
@@ -216,4 +216,15 @@ export const tests: ReadonlyArray<OneTest> = [
     expectedPath: join("/root", "mylib", "index.cjs"),
     extensions: defaultExtensionsWhenRunningInTsNode,
   },
+  {
+    name: "should resolve .ts from .js alias",
+    absoluteBaseUrl: "/root/",
+    paths: {
+      "@/*": ["src/*"],
+    },
+    existingFiles: [join("/root", "src", "foo.ts")],
+    requestedModule: "@/foo.js",
+    expectedPath: join("/root", "src", "foo.ts"),
```

Comment:
Will your resolution work if the path is selected as follows?
```suggestion
    expectedPath: join("/root", "src", "foo"),
```

If so this [PR](https://github.com/dividab/tsconfig-paths/pull/216) would address it as well - otherwise it does not.